src/writing-your-own-application/std-applications/understanding-esp-idf-template.md

```diff
@@ -0,0 +1,178 @@
+# Understanding esp-idf-template
+
+Now that we know how to [generate a std project], let's inspect what the generated project contains and try to understand every part of it.
+
+## Inspecting the generated Project
+
+When creating a project from [esp-idf-template] using:
+- MCU: `esp32c3`
+- ESP-IDF version: `v4.4`
+- STD support: `true`
+- Devcontainer support: `false`
+
+It should generate a file structure like this:
+
+```text
+├── build.rs
+├── .cargo
+│   └── config.toml
+├── Cargo.toml
+├── .gitignore
+├── rust-toolchain.toml
+├── sdkconfig.defaults
+└── src
+    └── main.rs
+```
+
+Before going further let's see what these files are for.
+
+- [.gitignore]
+    - tells `git` which folders and files to ignore
+- [Cargo.toml]
+    - the usual Cargo manifest declaring some meta-data and dependencies of the project
+- LICENSE-APACHE, LICENSE_MIT
+    - those are the most common licenses used in the Rust ecosystem
+    - if you want to apply a different license you can delete these files and change the license in `Cargo.toml`
+- [rust-toolchain.toml]
+    - defines which Rust toolchain to use
+    - depending on your target this will use `nightly` or `esp`
+- [.cargo/config.toml]
+    - the Cargo configuration
+    - contains our target
+    - contains `runner = "espflash --monitor"` - this means you can just use `cargo run` to flash and monitor your code
+    - contains the linker to use, in our case, [`ldproxy`]
+    - contains the unstable `build-std` cargo feature enabled.
+    - contains the `ESP-IDF-VERSION` envrionment variable that tells [`esp-idf-sys`] which ESP-IDF version the project will use.
+- src/main.rs
+    - the main source file of the newly created project
+    - we will examine its content in the next section
+- [build.rs]
+    - propagates linker arguments for `ldproxy`.
+- [sdkconfig.defaults]
+    - contains the overriden values from the ESP-IDF defaults.
+
+## `main.rs`
+
+```rust,ignore
+use esp_idf_sys as _; // If using the `binstart` feature of `esp-idf-sys`, always keep this module imported
+
+fn main() {
+    esp_idf_sys::link_patches();
+    println!("Hello, world!");
+}
+
+```
+The first line its an import that defines the esp-idf entry-point when the root crate is a binary crate that defines a main function.
+
+Then, we have an usual main function with two lines on it:
+- A call to `esp_idf_sys::link_patches` function that makes sure that a few patches to the ESP-IDF which are implemented in Rust are linked to the final executable.
+- We print in our console the famous "Hello World!".
+
+## Running the Code
+
+Building and running the code is as easy as
+
+```shell
+cargo run
+```
+
+This builds the code according to the configuration and executes [`espflash`] to flash the code to the board.
+
+Since our [`runner` configuration] also passes the `--monitor` argument to [`espflash`] we can see what the code is printing.
```

Comment:
The `espflash` links have no associated URL, and as such are rendering funny.

Reply:
Good catch, fixed!